File: src/Enterprise/C#/patterns/UnitOfWork.cs
using System.Collections.Generic;
using System.Linq;
namespace Genkidama.PatternExamples;
public static class UnitOfWorkExample { public static bool Run(){var store=new List<int>();var pending=new List<int>{2,3};store.AddRange(pending);pending.Clear();return store.SequenceEqual(new[]{2,3})&&pending.Count==0;} }

File: src/Enterprise/VB.NET/patterns/UnitOfWork.vb
Imports System.Collections.Generic
Imports System.Linq
Friend Module UnitOfWorkExample
    Friend Function Run() As Boolean
        Dim store As New List(Of Integer),pending As New List(Of Integer) From {2,3}
        store.AddRange(pending)
        pending.Clear()
        Return store.SequenceEqual({2,3}) AndAlso pending.Count=0
    End Function
End Module
